### ISSUE TITLE
'Ojbect reference not set to an instance of an object' when calling MeanShift.Learn

### ISSUE THREAD

**Issue by elJorno** [accord-net/framework#293]
**Stack Trace:** 
```
at Accord.MachineLearning.MeanShift.Learn(Double[][] x, Int32[] weights)
   at VST.OCR.OCREngine.<ReadProducts>d__6.MoveNext() in C:\Users\derp\Workspaces\Visual Studio\VST\OCR.cs:line 170
   at System.Linq.Enumerable.WhereEnumerableIterator``1.MoveNext()
   at VST.Form1.<>c__DisplayClass11_0.<ProcessFile>b__0() in C:\Users\derp\Workspaces\Visual Studio\VST\Form1.cs:line 152
```

**Code:**
```
var kernel = new GaussianKernel(dimension: 1);  // Same problem with UniformKernel
var go = new MeanShift(dimension: 1, kernel: kernel, bandwidth: 60);

var x = characterLocations.Select(l => new[] { (double)l }).ToArray();
go.Learn(x);
```

I tried passing a vector of ones for the weights, that gave me a slightly different stack trace, but also a NullReferenceException

I'm using the 3.2.0 release

**Comment by cesarsouza** [comment]
Thanks for reporting the issue! A new release will be made tomorrow fixing this and many other issues left over from the transition to the new API!

**Comment by cesarsouza** [comment]
Fixed on release 3.3.0.

**Comment by elJorno** [comment]
Wow, that was fast. Great work cesarsouza. I had a look myself but couldnt locate the problem immediately :)
